recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -0,0 +1,490 @@
+from conan import ConanFile
+from conan.tools.files import apply_conandata_patches
+from conan.tools.cmake import CMakeToolchain, cmake_layout
+from conans import CMake, tools
+import os
+import re
+
+
+def _strip_version(package):
+    return re.sub(r"/.*", "", package)
+
+
+def _option_name(dep):
+    if "option" in dep:
+        return dep["option"]
+    return "with_" + _strip_version(dep["require"])
+
+
+def _make_options(gdal_deps):
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "tools": [True, False]
+    }
+    for dep in gdal_deps:
+        if "require" in dep:
+            options[_option_name(dep)] = [True, False]
+    return options
+
+
+def _make_default_options(gdal_deps):
+    default_options = {"shared": False, "fPIC": True, "tools": False}
+    for dep in gdal_deps:
+        if "require" in dep:
+            default_options[_option_name(dep)] = ("default" in dep
+                                                  and dep["default"])
+
+    return default_options
```

Comment:
Lot of indirection, is it a good idea for maintainability?

Reply:
I think yes, because gdal has many dependencies that can all be treated almost the same.
Then, all dependency-specific information is stored at the same place.

But if you want me to change it, just let me know how you would like it.

Reply:
Having one recipe that is different from the other 1300+ is less nice... there are a few other trade offs like 
prevents the automatic merge of bump dependencies

Reply:
I think that many options properly handled in pre 3.5.0 are broken here since I don't see their library in requires (it's not straightforward to see this issue due to this indirection):
- gta
- hdf4
- heif
- kea
- mongocxx
- xerces
- webp
- podofo
- pcre
- pcre2
- odbc
- mysql
- freexl

Some other were not handled in pre 3.5.0, but at least there was a ConanInvalidConfiguration for them.

And finally some options in this recipe should be removed:
- libgeotiff & json-c are not optional for example. The underlying option in gdal build system has another meaning (external vs internal).

Reply:
I removed the indirection and tried to keep compatibility with pre-3.5.
I still have to take care of with_jpeg and with_mysql that are incompatible.

Also, I will rename with_crypto to with_openssl. I think it is confusing.